pkg/build/api/types.go

```diff
@@ -616,6 +616,9 @@ type JenkinsPipelineBuildStrategy struct {
 
 	// Jenkinsfile defines the optional raw contents of a Jenkinsfile which defines a Jenkins pipeline build.
 	Jenkinsfile string
+
+	// Env contains additional environment variables you want to pass into a build pipeline
+	Env []kapi.EnvVar
```

Comment:
it is not currently (the other env fields for other strategies don't support the full power either, we have a card for that).

i'm not sure if we can ever reasonably support the full power in this case, it would require us doing awkward things like resolving the value before sending the build object to the sync plugin, which i dislike (similar to the awkward things we're doing w/ imagestreams today).  (I would not want to make the sync plugin responsible for resolving the value)